src/collection.ts

```diff
@@ -975,12 +998,95 @@ export class Collection<TSchema extends Document = Document> {
    */
   async count(filter: Filter<TSchema> = {}, options: CountOptions = {}): Promise<number> {
     return executeOperation(
-      this.s.db.s.client,
-      new CountOperation(
-        MongoDBNamespace.fromString(this.namespace),
-        filter,
-        resolveOptions(this, options)
-      )
+      this.client,
+      new CountOperation(this.fullNamespace, filter, resolveOptions(this, options))
     );
   }
+
+  /**
+   * @internal
+   *
+   * Returns all search indexes for the current collection.
+   *
+   * @param indexName - Optional.  If specified, only indexes with matching index names will be returned.
+   * @param options - The options for the list indexes operation.
+   *
+   * @remarks Only available when used against a 7.0+ Atlas cluster.
+   */
+  listSearchIndexes(options?: ListSearchIndexesOptions): ListSearchIndexesCursor;
+  listSearchIndexes(indexName: string, options?: ListSearchIndexesOptions): ListSearchIndexesCursor;
+  listSearchIndexes(
+    indexNameOrOptions?: string | ListSearchIndexesOptions,
+    options?: ListSearchIndexesOptions
+  ): ListSearchIndexesCursor {
```

Comment:
Since the return value is the same for all these methods so we should just have the union type argument and optional options

Reply:
That's not quite the same.  Collapsing them into a single overload implies that `listSearchIndexes(AggregateOptions, AggregateOptions)` is a valid signature.  Overloading the method clarifies which combinations of arguments are valid.

Reply:
<img width="504" alt="Screenshot 2023-05-30 at 1 00 49 PM" src="https://github.com/mongodb/node-mongodb-native/assets/23407842/dbf1c3d7-4f8d-4717-bd15-08e0c4f629cb">
<img width="510" alt="Screenshot 2023-05-30 at 1 00 30 PM" src="https://github.com/mongodb/node-mongodb-native/assets/23407842/bcabf4c3-f5c5-4506-912b-e2cc27f67990">

Maybe this is my particular workflow, but I often peruse the overloads for methods by scrolling through the modal that pops up in vscode.  I think it's beneficial to provide clear intended usages.  I added an additional tsdoc comment for the new overload.

Also, `listSearchIndexes(options, options)` won't work because the actual pipeline stage sent will be:

```
{ $listSearchIndexes: { name: <aggregation options > } }
```

Reply:
Yea that's fair, I think the union of types is equally clear, the possible inputs are listed in one popup, no scrolling needed. 

> Also, listSearchIndexes(options, options) won't work because the actual pipeline stage sent will be: `{ $listSearchIndexes: { name: <aggregation options > } }`

From the option handling:
```js
   : typeof indexNameOrOptions === 'object'
        ? null
```
This case should set the name to `null` and then the pipeline will be constructed correctly in the cursor since we test for `name == null`.